pkg/webhook/cspc_operation.go

```diff
@@ -114,6 +121,99 @@ func getCommonPoolSpecs(cspcNew, cspcOld *apis.CStorPoolCluster) (*poolspecs, er
 	return commonPoolSpecs, nil
 }
 
+// getExpandedRaidGroups returns the raid groups only if pool expansion is done
+// by the users
+func getExpandedRaidGroups(
+	newPoolSpec *apis.PoolSpec, oldRaidGroups []apis.RaidGroup) []apis.RaidGroup {
+	expandedRgs := []apis.RaidGroup{}
+	for _, newRg := range newPoolSpec.RaidGroups {
+		isRaidGroupExist := false
+		for _, oldRg := range oldRaidGroups {
+			if IsRaidGroupCommon(oldRg, newRg) {
+				isRaidGroupExist = true
+				break
+			}
+		}
+		// If no common blockdevices exists in raid groups then newRg is
+		// added for expansion
+		if !isRaidGroupExist {
+			expandedRgs = append(expandedRgs, newRg)
+		}
+	}
+	return expandedRgs
+}
+
+func getNewBDsFromStripeSpec(oldRg, newRg apis.RaidGroup) []string {
+	mapOldRg := map[string]bool{}
+	bds := []string{}
+	for _, bd := range oldRg.BlockDevices {
+		mapOldRg[bd.BlockDeviceName] = true
+	}
+	for _, bd := range newRg.BlockDevices {
+		if !mapOldRg[bd.BlockDeviceName] {
+			bds = append(bds, bd.BlockDeviceName)
+		}
+	}
+	return bds
+}
+
+// getBDsFromRaidGroups return list of blockdevices from list of raid groups
+func getBDsFromRaidGroups(rgs []apis.RaidGroup) []string {
+	bds := []string{}
+	for _, rg := range rgs {
+		for _, bd := range rg.BlockDevices {
+			bds = append(bds, bd.BlockDeviceName)
+		}
+	}
+	return bds
+}
+
+// getOldCommonRaidGroups returns raidGroups that contains index matching of
+// oldRaidGroups and newRaidGroups. If oldRaidGroup doesn't exist on
+// newRaidGroup then return error
+func getOldCommonRaidGroups(oldPoolSpec,
```

Comment:
This function returns both new and old raid groups. Can we name it as `getIndexedCommonRaidGroups` ?

Reply:
Updated